fml/raster_thread_merger.cc

```diff
@@ -50,20 +50,36 @@ RasterThreadMerger::CreateOrShareThreadMerger(
 }
 
 void RasterThreadMerger::MergeWithLease(size_t lease_term) {
-  std::scoped_lock lock(mutex_);
   if (TaskQueuesAreSame()) {
     return;
   }
-  if (!IsEnabledUnSafe()) {
+  if (!IsEnabled()) {
     return;
   }
+
   FML_DCHECK(lease_term > 0) << "lease_term should be positive.";
+  {
+    std::scoped_lock lock(mutex_);
+    if (IsMergedUnSafe()) {
+      merged_condition_.notify_one();
+      return;
+    }
+  }
 
-  if (IsMergedUnSafe()) {
-    merged_condition_.notify_one();
-    return;
+  // Here we flush all expired tasks in the raster task queue before thread
+  // merging to ensure these sync tasks the platform thread post has been
+  // executed.
+  // TODO(0xZOne): Maybe we should only flush expired sync tasks posted by the
+  // platform thread.
+  if (MessageLoop::IsInitializedForCurrentThread()) {
+    if (!IsOnPlatformThread()) {
+      MessageLoop::GetCurrent().RunExpiredTasksNow();
+    }
```

Comment:
@iskakaushik @cyanglaz - do we want to specialize this to the PlatformThread? 

Do we not also need to flush any raster tasks?

Should we just be flushing both queues first?